src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -234,6 +234,9 @@ void RadioSpinel<InterfaceType, ProcessContextType>::Init(bool aResetRadio,
     if (aResetRadio)
     {
         SuccessOrExit(error = SendReset());
+#if OPENTHREAD_SPINEL_CONFIG_RESET_CONNECTION
+        SuccessOrExit(error = mSpinelInterface.ResetConnection());
```

Comment:
There is another call to `SendReset()` at <https://github.com/openthread/openthread/blob/60c15cea75556db4772a14762fcc6b57138376d9/src/lib/spinel/radio_spinel_impl.hpp#L2232>

There should also be a call to `ResetConnection()` there.